### PR TITLE
Return error, not 500, on AdminRestoreView error

### PR DESCRIPTION
**Pull request by kaapstorm**
Context: [FB 242755](http://manage.dimagi.com/default.asp?242755)

AdminRestoreView.get_context_data wasn't handling 500 responses.

@snopoke cc @sravfeyn @wspride buddy @biyeun

### CODE REVIEW

**Comment by czue** [comment]
I think this should be closed in favor of https://github.com/dimagi/commcare-hq/pull/14066

**Comment by czue** [comment]
just thinking that we probably do still want to get 500s when this page fails

**Comment by czue** [comment]
actually offlined with @kaapstorm - i think this is probably a good change but we might want to make sure we get notified on errors we don't expect. not sure if worth digging into that now or leaving as-is since this doesn't come up too frequently.

File: corehq/apps/hqadmin/views.py [hunk 1]
```diff
@@ -490,11 +492,18 @@ def get_context_data(self, **kwargs):
...
-        num_cases = len(xml_payload.findall('{http://commcarehq.org/case/transaction/v2}case'))
-        formatted_payload = etree.tostring(xml_payload, pretty_print=True)
+        if isinstance(response, StreamingHttpResponse):
+            string_payload = ''.join(response.streaming_content)
+            xml_payload = etree.fromstring(string_payload)
+            restore_id_element = xml_payload.find('{{{0}}}Sync/{{{0}}}restore_id'.format(SYNC_XMLNS))
+            num_cases = len(xml_payload.findall('{http://commcarehq.org/case/transaction/v2}case'))
+            formatted_payload = etree.tostring(xml_payload, pretty_print=True)
+        else:
+            # response is HttpResponse 500
```
Comment (czue) [inline corehq/apps/hqadmin/views.py]: we don't actually know this to be true right - just that it's an `HttpResponse`? could add a check for the status code or just update the comment to say something like "response is HttpResponse which is unexpected - tell the user something might have gone wrong".

Then I think we should also add a `messages.error` message along the lines of: "unexpected restore response - if you believe this is a bug please report an issue" so it's clear something has gone wrong to the user and they can do something about it.

**Comment by kaapstorm** [comment]
Reopening with better error message.

Example 401 response in a browser:
restore_response

File: corehq/apps/hqadmin/views.py [hunk 2]
```diff
@@ -490,11 +492,24 @@ def get_context_data(self, **kwargs):
...
-        formatted_payload = etree.tostring(xml_payload, pretty_print=True)
+        if isinstance(response, StreamingHttpResponse):
+            string_payload = ''.join(response.streaming_content)
+            xml_payload = etree.fromstring(string_payload)
+            restore_id_element = xml_payload.find('{{{0}}}Sync/{{{0}}}restore_id'.format(SYNC_XMLNS))
+            num_cases = len(xml_payload.findall('{http://commcarehq.org/case/transaction/v2}case'))
+            formatted_payload = etree.tostring(xml_payload, pretty_print=True)
+        else:
+            # get_restore_response return HttpResponse 401 (permissions), 404 (user not found) or 412
+            if response.status_code in (401, 404):
```
Comment (czue) [inline corehq/apps/hqadmin/views.py]: should `412` be in this list too based on above comment? that makes sense to me as those errors are also useful.

**Review by czue — APPROVED** [review]
looks good, one possible non-blocking followup

**Pull request merged**